src/tests/decoy_generator_test.py

```diff
@@ -0,0 +1,269 @@
+            530.23199463,
+            543.75042725,
+            588.25584555,
+            607.35174561,
+            661.27748266,
+            667.29205322,
+            704.27716064,
+            718.29813999,
+            764.34429932,
+            815.35154575,
+            824.4045909
+        ]
+    )
+    intensity = np.asarray(
+        [
+            1071.7286,
+            809.87616,
+            2428.9036,
+            7298.865,
+            8358.526,
+            1540.6752,
+            909.5406,
+            1430.8278,
+            6245.454,
+            4982.6943,
+            3986.77,
+            939.83044,
+            899.7238,
+            879.45825,
+            844.35614,
+            1710.6083,
+            2625.6973,
+            1187.6097,
+            1316.555,
+            1473.7188,
+            997.8472,
+            2542.6887,
+            2019.7664,
+            866.9728,
+            1725.27,
+            894.8671,
+            8904.243,
+            1648.2794,
+            3597.6963,
+            1302.7625,
+            3326.1228,
+            749.92834,
+            1150.5756,
+            2103.1558,
+            1408.3391,
+            1826.2734,
+            1224.997,
+            3090.4355,
+            1821.3624,
+            1061.0614,
+            1367.4617,
+            773.3596,
+            10000.0,
+            871.0255,
+            1167.3612,
+            3559.0688,
+            2279.5278,
+            5233.7705,
+            987.56683
+        ]
+    )
+
+    return MsmsSpectrum("HAHC[Carbamidomethyl]VTPGC[Carbamidomethyl]YYR",
+                                380.918862288762, 4, mz, intensity)
+
+def test_shuffle_and_reposition(original_spectrum, expected_decoy_spectrum):
+    # Get decoy
+    with unittest.mock.patch(
+            "ann_solo.config.config._namespace",
+            {"fragment_mz_tolerance": 10, "fragment_tol_mode": 'ppm'}) as _, \
+        unittest.mock.patch(
+            "decoy_generator._shuffle",
+            return_value=('HAHCVTPGCYYR', {0:4, 1:7, 2:2, 3:3, 4:10, 5:1, 6:6,
+                                           7:0, 8:8, 9:5, 10:9, 11:11})) as _:
+        decoy_spectrum = shuffle_and_reposition(original_spectrum)
+
+    # Assert that the shapes of of arrays are the same
+    assert np.shape(original_spectrum.mz) == np.shape(decoy_spectrum.mz)
+    assert np.shape(original_spectrum.intensity) == np.shape(
+        decoy_spectrum.intensity)
+
+    # Assertion of decoy M/Zs compared to original M/Zs
+    assert np.min(decoy_spectrum.mz) >= np.min(original_spectrum.mz)
+
+    # Assert that the sequences are not similar but are made of same residues
+    assert decoy_spectrum.proforma.sequence != original_spectrum.proforma.sequence
+    assert Counter(decoy_spectrum.proforma.sequence) == \
+           Counter(original_spectrum.proforma.sequence)
+
+    # Check if the generated spectrum is almost equal to the expected
+    np.testing.assert_allclose(original_spectrum.mz,

```

Comment:
**question:** I don't understand this test. If the fragments are repositioned, doesn't that change their m/z and thus this assertion will be False? Or should it be `decoy_spectrum.mz` instead of `original_spectrum.mz`?

Reply:
You are right, seems that the commit before my final change was the one captured in the final push. 
Should be changed to decoy_spectrum.mz and decoy_spectrum.intensity. I modified it.